### PR TITLE
checkstyle: move from individual modules to root poms

### PR DESCRIPTION
**Pull request by dhalperi**
Checkstyle config is no longer needed in individual modules;
instead, move it only to sdks/ , runners/, and examples/ poms.
Individual modules like sdks/java/core may still reference checkstyle
to configure their own options.

The only sketchy bit is that I had to link sdks/java/build-tools to
beam-parent rather than sdks/java so that I didn't create a build
dependency loop.

### CODE REVIEW

**Comment by dhalperi** [comment]
R: @davorbonaci

File: pom.xml [hunk 1]
```diff
@@ -138,6 +138,9 @@
   <packaging>pom</packaging>
 
   <modules>
+    <!-- sdks/java/build-tools has project-wide configuration. To make these available
+      in all modules, link it directly to the parent pom.xml. -->
+    <module>sdks/java/build-tools</module>
```
Comment (swegner) [inline pom.xml]: Should `build-tools` move to the top-level?

Comment (davorbonaci) [inline pom.xml]: Too many stuff at top-level makes it harder to navigate. Then, these are java-build tools, not so much Python or other SDKs.

**Comment by davorbonaci** [comment]
LGTM. Please merge once tests pass.

**Pull request closed without merging**